src/dmd/expressionsem.d

```diff
@@ -2308,6 +2308,35 @@ private bool functionParameters(const ref Loc loc, Scope* sc,
     return (err || olderrors != global.errors);
 }
 
+/**
+ * Determines whether a `Type` represents a module or package
+ * (Used as a helper for is(type == module) and is(type == package))
+ *
+ * Params:
+ *  ty: the `Type` to be checked
+ *
+ * Returns:
+ *  the `Package` which the `Type` represents (or `null` if it doesn't represent a `Package`)
+ */
+private Package typeIsPackage(Type ty, Scope* sc)
```

Comment:
This function can be better reused if it just receives the `Dsymbol` while using `ty.toDsymbol(sc)` at the callsite. You also don't need the undocumented `sc` parameter then.